x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_icon_property.tsx

```diff
@@ -115,21 +117,26 @@ export class DynamicIconProperty extends DynamicStyleProperty<IconDynamicOptions
   renderLegendDetailRow({ isPointsOnly, isLinesOnly }: LegendProps) {
     const { stops, fallbackSymbolId } = this._getPaletteStops();
     const breaks = [];
+    const layerStyle = this._layer.getCurrentStyle() as IVectorStyle;
     stops.forEach(({ stop, style }) => {
       if (stop) {
+        const { svg } = layerStyle.getIconMeta(style);
```

Comment:
change to `const svg = layerStyle.getIconMeta(style)?.svg;`, getIconMeta can return undefined